compiler/rustc_passes/src/check_attr.rs

```diff
@@ -619,6 +619,9 @@ impl CheckAttrVisitor<'_> {
             }
             // we check the validity of params elsewhere
             Target::Param => return false,
+            Target::Expression => Some("expression"),
+            Target::Statement => Some("statement"),
+            Target::Arm => Some("match arm"),
             _ => None,
```

Comment:
Could you make the match exhaustive?  Probably refactor the `Display` impl for `Target` to avoid manually copying the names here.

Reply:
Good idea. I just added commits to resolve those issues.